pkg/controller/stacks/install/installjob.go

```diff
@@ -328,3 +331,49 @@ func isCRDObject(obj runtime.Object) bool {
 	return apiextensions.SchemeGroupVersion == gvk.GroupVersion() &&
 		strings.EqualFold(gvk.Kind, "CustomResourceDefinition")
 }
+
+func setStackDefinitionControllerEnv(obj *unstructured.Unstructured, namespace, name string) error {
+	env := []corev1.EnvVar{{
+		Name:  stacks.StackDefinitionNamespaceEnv,
+		Value: namespace,
+	}, {
+		Name:  stacks.StackDefinitionNameEnv,
+		Value: name,
+	}}
+
+	// use convert functions because SetUnstructuredContent is unwieldy
+	sd, err := convertToStackDefinition(obj)
+	if err != nil {
+		return err
+	}
+
+	ts := &sd.Spec.Controller.Deployment.Spec.Template.Spec
```

Comment:
Could `Spec.Controller` be `nil` too?

Reply:
@displague just a general style suggestion, it's easier to follow the code if you check for error and return early instead of putting the logic inside the if block. See https://danp.net/2015/11/02/reducing-go-nesting.html